cpp/src/spatial/hausdorff.cu

```diff
@@ -37,90 +39,20 @@ namespace {
 
 using size_type = cudf::size_type;
 
-struct hausdorff_index {
-  int32_t space_row;
-  int32_t space_col;
-  int32_t point_row;
-  int32_t point_col;
-  int32_t result_idx;
-};
-
-/** @brief Traverses a cartesian product of indices.
- *
- * Traverses a cartesian product of indices such that pairs of points within a pair of spaces appear
- * consecutively. This is used in a transform iterator to calculate multiple hausdorff distances
- * simultaneously by producing distances in the correct order for consumption by `reduce_by_key`.
- * The reduce key is the pair of spaces. The pair of points within a pair of spaces appears
- * consecutively.
- *
- */
-template <typename SpaceSizeIterator>
-struct haus_traversal_functor {
-  int32_t num_spaces;
-  int32_t num_points;
-  size_type const* space_offsets;
-  SpaceSizeIterator const space_sizes;
-
-  int32_t __device__ space_lookup(int32_t point_idx)
-  {
-    return thrust::upper_bound(thrust::seq, space_offsets, space_offsets + num_spaces, point_idx) -
-           space_offsets - 1;
-  }
-
-  hausdorff_index __device__ operator()(int32_t const idx)
-  {
-    int32_t const space_a_idx      = space_lookup(idx / num_points);
-    int32_t const space_a_offset   = space_offsets[space_a_idx];
-    int64_t const space_a_offset_n = space_a_offset * static_cast<int64_t>(num_points);
-    int32_t const space_a_size     = space_sizes[space_a_idx];
-
-    int32_t const space_b_idx    = space_lookup((idx - space_a_offset_n) / space_a_size);
-    int32_t const space_b_offset = space_offsets[space_b_idx];
-    int32_t const space_b_size   = space_sizes[space_b_idx];
-
-    int32_t const space_begin = space_a_offset_n + space_a_size * space_b_offset;
-    int32_t const cell_idx    = idx - space_begin;
-    int32_t const cell_col    = cell_idx / space_b_size;
-
-    int64_t const source_idx =
-      space_a_offset_n + space_b_offset + (num_points - space_b_size) * cell_col + cell_idx;
-    int32_t const source_col = source_idx / num_points;
-    int32_t const source_row = source_idx % num_points;
-
-    int32_t const destination_idx =
-      space_a_offset_n + (space_a_size - 1) * num_points + space_b_offset + (space_b_size - 1);
-
-    int32_t const result_idx =
-      destination_idx == source_idx ? space_a_idx * num_spaces + space_b_idx : -1;
-
-    return {
-      space_a_idx,
-      space_b_idx,
-      source_row,
-      source_col,
-      result_idx,
-    };
-  }
-};
-
 template <typename T>
-struct hausdorff_index_to_acc_functor {
+struct gcp_to_hausdorff_acc {
```

Comment:
What is gcp? Please avoid abbreviation unless it's obvious. "acc" is also ambiguous, but I let it slide before. :)

Reply:
`cartesian_product_group_index_to_hausdorff_accumulator_functor`

In the immortal words of @jrhemstad,..

> Naming is hard yo

I'll try something else, but higher order functions are really difficult to name. 😞 

Any suggestions?

Reply:
`hausdorfff_accumulator_factory` is the best I've come up with so far. :man_shrugging: